src/transport/raw/TCP.cpp

```diff
@@ -617,56 +617,53 @@ void TCPBase::HandleIncomingConnection(const Inet::TCPEndPointHandle & listenEnd
                                        const Inet::IPAddress & peerAddress, uint16_t peerPort)
 {
     TCPBase * tcp = reinterpret_cast<TCPBase *>(listenEndPoint->mAppState);
-    // To handle errors around endpoint in the calls to HandleAcceptError, we must initialize mAppState
-    endPoint->mAppState = tcp;
+    ReturnAndLogOnFailure(tcp->DoHandleIncomingConnection(listenEndPoint, endPoint, peerAddress, peerPort), Inet,
+                          "Failure accepting incoming connection");
+}
 
+CHIP_ERROR TCPBase::DoHandleIncomingConnection(const Inet::TCPEndPointHandle & listenEndPoint,
+                                               const Inet::TCPEndPointHandle & endPoint, const Inet::IPAddress & peerAddress,
+                                               uint16_t peerPort)
+{
     PeerAddress addr;
-    CHIP_ERROR err = GetPeerAddress(*endPoint, addr);
-    VerifyOrReturn(err == CHIP_NO_ERROR, HandleAcceptError(endPoint, err));
-    ActiveTCPConnectionState * activeConnection = tcp->AllocateConnection(endPoint, addr);
-    if (activeConnection != nullptr)
+    ReturnErrorOnFailure(GetPeerAddress(*endPoint, addr));
```

Comment:
Same here - terrible diff, I'm not sure why; the only thing that's changed is:

1) This is no longer a static method so we don't need `tcp->`
2) Early-return `CHIP_ERROR_TOO_MANY_CONNECTIONS` instead of wrapping everything in `if {}`